design/velero-client-download-apiserver_design.md

```diff
@@ -0,0 +1,105 @@
+# Velero Endpoint Download Client
+
+## Abstract
+
+<!-- One to two sentences that describes the goal of this proposal and the problem being solved by the proposed change.
+The reader should be able to tell by the title, and the opening paragraph, if this document is relevant to them. -->
+Velero data is stored in an object store.
+When CLI users want to download data from the object store, it connects to the specific object store directly to download data.
+This assumes the target storing the data is always accessible from the client side. This is not always true, especially in the on premise environments.
+Velero may also save the data in non object stores in the future so an object store may not be available for the CLI to download from.
+
+This design proposes to add a new endpoint on the Velero server that CLI users can connect to and download data from without having to connect to an object store directly.
+
+## Background
+Velero CLI commands such as following download data from the object store directly today.
+```
+velero backup download
+velero describe
+velero backup logs
+...
+```
+We want them to work even when the object store is not accessible from the client side.
+Even if the object store is accessible from the client side, today things like cacert and insecure-skip-tls-verify flags are specified manually from the client even though the Velero server already has the information. This work eliminates the client from having to know about the object store configuration and certificates.
+## Goals
+- Enable Velero CLI users to download data from the Velero server without having to connect to the object store directly for scenarios where the object store is not accessible from the client side.
+
+## Non Goals
+- Making the download server the default way to download data from object store.
+
+## High-Level Design
+Create an endpoint on the Velero server that CLI users can connect to and download data from without having to connect to the object store directly.
```

Comment:
Can we highlight that we will still keep the direct download way for most user cases and only download data via the download server for the BSLs that are configured to use it explicitly? We do this because the ingress/service approach involves some extra configuration items for users and we don't want to change the UX for most users.